### PR TITLE
feat: Add upcoming events to the home page

### LINKED ISSUE #166: [Feature Request] Upcoming Events on Home Page
### Overview

Add up to three upcoming events on home page.

### Details

See attached screenshot.

If there and no events, hide this component completely. 

### Why would this feature be useful?

So that people can more easily see upcoming CS Club events.

### Additional Notes

image

### PR DESCRIPTION
**Pull request by codingzilin**
### Description
Added upcoming events for home page.

### Changes Made

- Upcoming events card.
- Added conditional rendering for the 'Upcoming Events' section.
- Included a map function to iterate over UPCOMING_EVENTS and render UpcomingEventCard components."

### Related Issues
Fixes #166 

### Additional Notes
N/A

### CODE REVIEW

File: package.json
Comment (phoenixpereira) [inline package.json:1]: Please keep the package version the same to ensure consistency.

File: src/app/(home)/page.tsx [hunk 1]
```diff
@@ -2,11 +2,54 @@ import Duck from '@/components/Duck';
...
+                                <FiMapPin size={26} />
+                                <span>{event.location}</span>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </FancyRectangle>
+    );
+}
```
Comment (phoenixpereira) [inline src/app/(home)/page.tsx:43]: Make this a seperate component by moving it to a new file within the `(home)` folder.

File: src/app/(home)/page.tsx [hunk 2]
```diff
@@ -62,36 +105,63 @@ export default function HomePage() {
...
+                    <div
+                        className={`flex flex-col justify-around ${UPCOMING_EVENTS.length === 0 ? 'lg:col-span-2' : ''}`}
+                    >
+                        <div>
+                            <div className="relative z-10 mt-12 flex flex-col text-2xl font-black md:flex-row lg:mt-24 lg:text-3xl">
+                                <h3>New Members are</h3>
+                                <div className="mt-2 w-fit bg-purple px-2 md:ml-2 md:mt-0">
+                                    <h3 className=" text-grey">Always Welcome</h3>
+                                </div>
+                            </div>
```
Comment (phoenixpereira) [inline src/app/(home)/page.tsx:118]: Make "New Members are" and "Always Welcome" go on seperate lines like this:

image

File: src/app/(home)/page.tsx [hunk 3]
```diff
@@ -2,11 +2,54 @@ import Duck from '@/components/Duck';
...
+                                {event.title}
+                            </h4>
+                            <div className="flex gap-2">
+                                <FiClock size={26} />
+                                <span>{event.time}</span>
+                            </div>
+                            <div className="flex gap-2">
+                                <FiMapPin size={26} />
+                                <span>{event.location}</span>
+                            </div>
```
Comment (phoenixpereira) [inline src/app/(home)/page.tsx:36]: Make the event time and location text smaller and not bold by adding `text-lg font-normal` to the `className`. Doing this should help fix the icons being different sizes.

File: src/app/(home)/page.tsx [hunk 4]
```diff
@@ -2,11 +2,54 @@ import Duck from '@/components/Duck';
...
+                        </div>
```
Comment (phoenixpereira) [inline src/app/(home)/page.tsx:37]: Fix the icons being different sizes. This is caused by the event location being two lines while event time is only one.
image

File: src/app/(home)/page.tsx [hunk 5]
```diff
@@ -62,36 +105,63 @@ export default function HomePage() {
...
-                <div className="relative z-10 mt-4 border-2 border-white bg-grey px-4 py-4 md:px-6 md:py-6">
-                    <p className="text-lg md:text-xl">
-                        Are you a first year student? The Club runs activities at the start of the
-                        year specifically for you, giving you a chance to meet other students, and
-                        helping you ease into uni life.
-                    </p>
+
+                    {/* <div className="relative z-10 mt-12 text-2xl font-black md:flex-row lg:ml-4 lg:mt-24 lg:text-3xl"> */}
+                    {UPCOMING_EVENTS.length > 0 && (
+                        <div className="relative z-10 mt-12 text-2xl font-black md:flex-row lg:ml-4 lg:mt-24 lg:text-3xl">
```
Comment (phoenixpereira) [inline src/app/(home)/page.tsx:152]: Add a bit more horizontal gap between this and the New Members are Always Welcome section by changing `lg:ml-4` to `lg:ml-10`

File: src/app/(home)/page.tsx [hunk 6]
```diff
@@ -62,36 +105,63 @@ export default function HomePage() {
...
+                            <div className="flex justify-end">
```
Comment (phoenixpereira) [inline src/app/(home)/page.tsx:153]: The title shouldn't be justified to the end on mobile.

**Review by phoenixpereira — CHANGES_REQUESTED** [review]

**Comment by codingzilin** [comment]
Fixed changes required

**Review by phoenixpereira — CHANGES_REQUESTED** [review]
Just some small changes needed.

**Comment by codingzilin** [comment]
Thx Phoenix, fixed changes required.

**Review by phoenixpereira — APPROVED** [review]

**Review by rayokamoto — APPROVED** [review]

**Pull request merged**